GitCommands/UserRepositoryHistory/LocalRepositoryManager.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+
+namespace GitCommands.UserRepositoryHistory
+{
+    /// <summary>
+    /// Manages the history of local git repositories.
+    /// </summary>
+    public sealed class LocalRepositoryManager : IRepositoryManager
+    {
+        private const string KeyRecentHistory = "history";
+        private readonly IRepositoryStorage _repositoryStorage;
+
+        public LocalRepositoryManager(IRepositoryStorage repositoryStorage)
+        {
+            _repositoryStorage = repositoryStorage;
+        }
+
+        /// <summary>
+        /// <para>Saves the provided repository path to history of local git repositories as the "most recent".</para>
+        /// <para>If the history contains an entry for the provided path, the entry is physically moved
+        /// to the top of the history list.</para>
+        /// </summary>
+        /// <remarks>
+        /// The history is loaded from the persistent storage to ensure the most current version of
+        /// the history is updated, as it may have been updated by another instance of GE.
+        /// </remarks>
+        /// <param name="repositoryPath">A repository path to be save as "most recent".</param>
+        /// <returns>The current version of the history of local git repositories after the update.</returns>
+        /// <exception cref="ArgumentException"><paramref name="repositoryPath"/> is <see langword="null"/> or <see cref="string.Empty"/>.</exception>
+        /// <exception cref="NotSupportedException"><paramref name="repositoryPath"/> is a URL.</exception>
+        [ContractAnnotation("repositoryPath:null=>halt")]
+        public Task<IList<Repository>> AddAsMostRecentAsync(string repositoryPath)
+        {
+            if (string.IsNullOrWhiteSpace(repositoryPath))
+            {
+                throw new ArgumentException(nameof(repositoryPath));
+            }
+
+            if (PathUtil.IsUrl(repositoryPath))
+            {
+                // TODO: throw a specific exception
+                throw new NotSupportedException();
+            }
+
+            repositoryPath = repositoryPath.Trim()
+                                           .ToNativePath()
+                                           .EnsureTrailingPathSeparator();
+            return AddAsMostRecentRepositoryAsync(repositoryPath);
+
+            Task<IList<Repository>> AddAsMostRecentRepositoryAsync(string path)
+            {
+                return Task.Run<IList<Repository>>(async () =>
```

Comment:
❗️ This call to `Task.Run` is unnecessary. The enclosing local function is already asynchronous, so do this instead:

1. Mark the local function `async`
2. Remove the `Task.Run` wrapper
3. Switch to a background thread at the beginning of the function

    ```csharp
    await TaskScheduler.Default;
    ```